[Khoo Jing Xiang] iP

meowwtama opened the pull request:
### Anita MaxWynn

Do you want to:
- Win
- Win big
- Max win? :money_mouth_face:

Well, you certainly want to right? :sunglasses: To:
1. Buy a car,
2. Buy a condo
3. Have **_CASH_** :money_mouth_face: :money_mouth_face:

Use the `max_win()` to win big!!

THEN HERE ARE THE THINGS YOU NEED TO DO! :money_mouth_face: :money_mouth_face:
- [ ] ANITA
- [ ] MAX
- [ ] WYNN

>[!CAUTION]
Nothing worth having comes easy. So always remember:
```java
System.out.println("ALWAYS MAX WIN!!");
``` 

To learn more about our saviour, click [**_HERE_**](https://youtu.be/_NlVJeDGHVk?si=kiAu51WOoh_UzI7n)!!!!!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

meowwtama committed:
Level 0. Rename, Greet, Exit

Added the greet() and line() methods.

meowwtama committed:
Level 1. Echo

Added echo() method. Edited greet() by removing the "Bye" print statement.

meowwtama committed:
Level 2. Add, List

Updated echo() method to store the commands given by user instead of immediately outputting it console. Can now list out all the stored commands when user inputs "list".

meowwtama committed:
Level 3. Mark as Done

Created class Task which contains methods related to each Task. Removed static field commandList from class Duke. Edited echo() method in class Duke to account for commands mark and unmark.

meowwtama committed:
Level 4. ToDos, Events, Deadlines

Created 3 new classes Deadline, Todo, and Event. All 3 classes inherit from Task. Now able to add tasks of the 3 classes and updated the list command to factor in the new classes.

meowwtama committed:
Level 5. Handle Errors

Added error handling for invalid inputs, including no input, non-task related inputs, and missing information for various task inputs.
Added bye() function for Duke fike.

meowwtama committed:
Level 6. Delete

Added delete functionality with error handling for missing index.
Moved error handling to echo() function.

meowwtama committed:
Updated input and output for runtest.sh

added input and outputs to test for levels 1-6

meowwtama committed:
Add Database.java

Now able to store list data from previous sessions using a text file.

meowwtama committed:
Level 8. Dates and Times

Deadlines and events now accept dates in a format such as yyyy-mm-dd format (e.g., 2019-10-15) and print in a different format such as MMM dd yyyy e.g., (Oct 15 2019).

meowwtama committed:
A-MoreOOP

Added classes Parser, TaskList, and Ui to better implement OOP. Changed the organisation of various functions in the other classes as well.

meowwtama committed:
A-Packages

Added every class to anita package and reorganised import statements.

meowwtama committed:
Merge remote-tracking branch 'origin/add-gradle-support'

meowwtama committed:
A-JUnit

Added JUnit tests for Todo event and parser

meowwtama committed:
A-JavaDoc

Added javadocs for the following classes:
- Duke
- Parser
- Task
- TaskList

meowwtama committed:
A-CodingStandard update

Changed intellij settings to suit coding standards. Applied the changes to code. (eg. switch statements)

meowwtama committed:
Merge branch 'tmp'
commit is needed to fix detached head branch.

meowwtama committed:
Update TaskList.java

benson1029 commented on src/main/java/anita/Database.java:
```diff
@@ -0,0 +1,84 @@
+import java.util.Scanner;
+
+public class Database {
+    private File myFile;
```
`file` is already OK as variable name. `myFile` looks redundant.

benson1029 commented on src/main/java/anita/Database.java:
```diff
@@ -0,0 +1,84 @@
+
+public class Database {
+    private File myFile;
+    private Duke duke;
```
Better to change the class name and variable name to `Anita` and `anita` respectively.

benson1029 commented on src/main/java/anita/Database.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    public ArrayList<String> readFile() {
+        ArrayList<String> data = new ArrayList<>();
```
`data` is not very descriptive. Consider using `fileData`?

benson1029 commented on src/main/java/anita/Database.java:
```diff
@@ -0,0 +1,84 @@
+            String[] tokens = i.split("/");
+            String command = tokens[0].split(" ")[0];
+            try {
+                duke.addingTask(command, i);
```
Use `addTask` instead of `addingTask`.

benson1029 commented on src/main/java/anita/Database.java:
```diff
@@ -0,0 +1,84 @@
+
+    public void deleteLine(int index) {
+        ArrayList<String> data = readFile();
+        data.remove(index-1);
```
Add space before and after the `-` operator.
```suggestion
        data.remove(index - 1);
```

benson1029 commented on src/main/java/anita/Duke.java:
```diff
@@ -0,0 +1,106 @@
+import java.io.IOException;
+
+public class Duke {
+    protected static boolean initialize = true;
```
Use `isInitialized` instead of `initialize`.

benson1029 commented on src/main/java/anita/Duke.java:
```diff
@@ -0,0 +1,106 @@
+     * Main driver code for executing commands.
+     * If user command is not valid, throws the respective exception.
+     *
+     * @throws IOException
```
Try to describe in what cases `IOException` will be thrown.

benson1029 commented on src/main/java/anita/Duke.java:
```diff
@@ -0,0 +1,106 @@
+        Parser parser = new Parser();
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+
+        boolean isRunning = true;
```
This boolean naming is good.

benson1029 commented on src/main/java/anita/Duke.java:
```diff
@@ -0,0 +1,106 @@
+     * @param description The details of the task.
+     * @throws Exception If command is not valid.
+     */
+    public void addingTask(String command, String description) throws Exception {
```
Better to use the specific exception type instead of throwing `Exception`.

benson1029 commented on src/main/java/anita/Event.java:
```diff
@@ -0,0 +1,20 @@
+package anita;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+public class Event extends Task {
```
Add a blank line in between.

benson1029 commented on src/main/java/anita/Parser.java:
```diff
@@ -0,0 +1,99 @@
+        String[] tokens2 = tokens[0].split(" ", 2);
+        String[] res = new String[2];
+        String taskName = "";
+        if (tokens2.length == 1)
+            throw new ArrayIndexOutOfBoundsException("The description of a deadline cannot be empty.");
```
Use Egyptian style brackets.

benson1029 commented on src/main/java/anita/Parser.java:
```diff
@@ -0,0 +1,99 @@
+     * Custom parser for the Event task.
+     *
+     * @param description The raw user input.
+     * @return A String[] containing parts of the user input to be used for instantiation.
```
Can put `String[]` in `<code>` and `</code>`.

benson1029 commented on src/main/java/anita/Task.java:
```diff
@@ -0,0 +1,25 @@
+     * @return 'X' if task is completed else " ".
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
```
```suggestion
        return (isDone) ? "X" : " ";
```

benson1029 commented on src/main/java/anita/TaskList.java:
```diff
@@ -0,0 +1,93 @@
+     * @return Task to be retrieved.
+     */
+    public Task get(int index) {
+        return taskList.get(index -1);
```
Add space after the operator `-`.

benson1029 commented on src/main/java/anita/Todo.java:
```diff
@@ -0,0 +1,12 @@
+    public Todo(String description) {
+        super(description);
+    }
+    @Override
```
Add a blank line between the two methods.

benson1029 commented on src/main/java/anita/Ui.java:
```diff
@@ -0,0 +1,53 @@
+        System.out.println("What can I do for you?");
+        line();
+    }
+    public void bye() {
```
Add a blank line between the two methods.

benson1029 commented on src/main/java/anita/Ui.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    public void removeTaskMessage(Task task) {
+        System.out.println("Noted. I've removed this task: ");
```
Consider putting these magic literals as constants.

benson1029 reviewed (commented):
Good work. Some naming and coding standard issues to be fixed.

meowwtama committed:
Complete Level 9 find

Add case for "find" in Duke class.

Add findTask method in TaskList class.

Users can now make use of the find command to find tasks containing a certain keyword.

meowwtama committed:
Add checkstyle

Added check style files and dependencies.

Edited all java files such that it now follows the coding convention.

meowwtama committed:
Add Level 10

Added fxml dependencies and files.

Created and modified GUI for Anita chatbot.

meowwtama committed:
Add assertion

Duke, Task, TaskList classes: add assert calls.

Assertion calls can help to identify mistakes made in logic.
Useful to have some simple checks to catch mistakes.

Add assertion calls to check for null objects and range of index.

Done using assert statements to throw assertions when running.

meowwtama committed:
Make method private

Some methods in the codebase are unnecessarily written as public.

To enhance code maintainability and prevent unintended modification or misuse of internal functionality.
Restricting access to these methods can enforce better encapsulation and reduce the risk of introducing unexpected behaviour during future modifications.

Identified public methods get(int index) in TaskList class and getStatusIcon() in the Task abstract class are being refactored to private access modifiers.

Making these methods private encapsulates their behaviour, limiting their visibility and access to other parts of the codebase.

meowwtama committed:
Merge pull request #2 from meowwtama/A-Assertions

Add assertion

meowwtama committed:
Merge branch 'master' into A-CodeQuality

meowwtama committed:
Merge pull request #3 from meowwtama/A-CodeQuality

Make method private

meowwtama committed:
Update TaskList.java

Added getLast() method for future expansions

meowwtama committed:
Add Friendlier syntax

Added "t", "d", and "e" commands as shortcuts for "todo", "deadline", and "event" respectively.

meowwtama committed:
Enhance GUI

Added border boxes and a minimum size around the text chat to simulate a live chat.

Remove height restriction on dialogue box to allow all tasks to be shown without being cut off.

Changed font of chat to fix logo.

Increased length of ui.line() to fit the GUI dialogue box.

Removed calls for ui.line() in main file.

meowwtama committed:
Fix GUI

Added title for GUI and updated shadowjar.

meowwtama committed:
Fix mark and unmark task

Fixed a bug where the new status of the marked and unmarked tasks was not saved to the database.

This caused the original status to be lost and replaced with a default "false".

Fixed by changing the representation of saved data in the database.

Also changed the syntax for loading and unloading data.

Added new function Database.changeLine() to allow for the changing of status within the database.

meowwtama committed:
Update Readme.md

Updated the user guide portion to allow users to learn about various functionalities of the bot.

meowwtama committed:
Update README.md

meowwtama committed:
Fix javadoc for methods

Added fixes for javadocs comments.

meowwtama committed:
Add Unit test for Anita

Added more unit testing for allocateTask function.

meowwtama committed:
Update README.md

Formatted README.md to be more user friendly.

meowwtama committed:
Fix Database file create method

Previously database did not check if directory exists.

meowwtama committed:
Update README.md
